test/validation/api/pktio/pktio.c

```diff
@@ -276,6 +276,7 @@ static void pktio_init_packet_eth_ipv4(odp_packet_t pkt, uint8_t proto)
 
 	/* IP */
 	odp_packet_l3_offset_set(pkt, ODPH_ETHHDR_LEN);
+	odp_packet_has_ipv4_set(pkt, true);
```

Comment:
I think setting of the metadata required by TX checksumming should be done only for the test cases that do TX checksumming and not globally for all tests. If TX checksum is not requested, the L3 & L4 type metadata should not be needed by the ODP implementation being tested, so adding the metadata in that case can reduce the effectiveness of the API validation test.

Reply:
PKT metadata like l3 offset, l4 offset  are already used for checksum fixing up (in non check sum test cases) and also in 
pktio_pkt_set_seq() sequence number. Now it is just in addition to existing metadata we are also updating L3 and L4 type meta data. Since even l3 and l4 offsets are also not needed in non Tx HW checksum generation test cases, just removing L3 and L4 type from common packet initialization is also not full solution if we want to test without pkt metadata.

Reply:
Even if the offsets are now set even when they are not strictly needed for the ODP API under test, we do not need to do the same for the packet types. As far as I can see it would be sufficient to set the types in pktio_test_chksum() and pktio_test_chksum_sctp() instead of the generic packet creation functions to restrict the type metadata only to the checksum tests. So why not do it that way?

Reply:
Ok. If setting them outside is fine, then I'm fine. I was only thinking it might be out of context if we don't set it at packet creation. Other way is propagate fix_cs flag upto packet creation and then set it and clear it based on whether we are doing HW checksum offload or not. This second approach would involve changes at lot of places as it is generic packet creation code.

Reply:
There is no need to set the flags in packet creation and unnecessarily complicate the code. The flags are needed only by checksum offload, so they are related to checksumming in the context of this application and are thus not out of place in the checksum test functions.